heron/spi/src/java/com/twitter/heron/spi/packing/PackingPlanProtoDeserializer.java

```diff
@@ -0,0 +1,60 @@
+// Copyright 2016 Twitter. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.twitter.heron.spi.packing;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.twitter.heron.proto.system.PackingPlans;
+
+/**
```

Comment:
I've gone back and forth on this in the past. In other projects I've implemented both in one class before finding that most clients rely on one and not the other, and sometimes each has different deps. Hence I implemented them separately, which also keeps the class size small and easy to read.